pkg/logging/fields.go

```diff
@@ -10,12 +10,14 @@ import (
 	"go.uber.org/zap/zapcore"
 )
 
-var EntryMessageField = "entryMessage"
-
 type fileField struct {
 	f core.File
 }
 
+type LogMessageProducer interface {
+	GetMessage(entry zapcore.Entry) string
+}
```

Comment:
It's semi-related to auth, in that this PR adds a second implementation of the "send this string instead of the sanitized json" (`unsanitizedMessage)`. That implementation doesn't send the message, but rather a separate bit of text that's designed for analytics (not for human readability on the console).

From your comment below, it sounds like you actually don't like drawing that distinction, and would prefer that the message on the console play double-duty and also be the message we send to analytics. Is that right? If so, I can just remove this code.

Relates to #239

Reply:
> From your comment below, it sounds like you actually don't like drawing that distinction, and would prefer that the message on the console play double-duty and also be the message we send to analytics. Is that right?

The original intent, and my thinking is more like "anything important enough to go to console should go, sanitized, to analytics". There's also stuff that should go to analytics that doesn't go to console.

The second part of my comment was acknowledging that the caller may want to control how the message is sanitized. It's the transformation rather than production aspect that I'm stuck on.

> If so, I can just remove this code.

I don't mean to present as the authority on it. I don't object enough to block anything or require and immediate action.

Reply:
For the sake of consistency I'll remove this. I do think that we should separate "what we send to analytics" from "what we show to the user", but that's #239. So I'll go with prior art here, and we can revisit it or not as part of that ticket.